src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATaskTests.cs

```diff
@@ -741,5 +741,202 @@ async Task CoreAsync()
+            sum += item;
+        return sum;
     }
-}
\ No newline at end of file
+}
+";
+
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Default.AddPackages(
+                    ImmutableArray.Create(new PackageIdentity("Microsoft.Bcl.AsyncInterfaces", "5.0.0"))),
+                LanguageVersion = CSharpLanguageVersion.CSharp8,
+                TestCode = code,
+                FixedCode = fixedCode,
+            }.RunAsync();
+        }
+
+        [Fact, WorkItem(3703, "https://github.com/dotnet/roslyn-analyzers/issues/3703")]
+        public async Task CSharpAwaitForEachWithCancellation()
+        {
+            var code = @"
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+public static class AsyncExtensions
+{
+    public static async Task<int> Sum(this IAsyncEnumerable<int> items, CancellationToken cancellationToken)
+    {
+        int sum = 0;
+        await foreach (var item in [|items.WithCancellation(cancellationToken)|])
+            sum += item;
+        return sum;
+    }
+}
+";
+            var fixedCode = @"
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+public static class AsyncExtensions
+{
+    public static async Task<int> Sum(this IAsyncEnumerable<int> items, CancellationToken cancellationToken)
+    {
+        int sum = 0;
+        await foreach (var item in items.WithCancellation(cancellationToken).ConfigureAwait(false))
+            sum += item;
+        return sum;
+    }
+}
+";
+
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Default.AddPackages(
+                    ImmutableArray.Create(new PackageIdentity("Microsoft.Bcl.AsyncInterfaces", "5.0.0"))),
+                LanguageVersion = CSharpLanguageVersion.CSharp8,
+                TestCode = code,
+                FixedCode = fixedCode,
+            }.RunAsync();
+        }
+
+        [Fact, WorkItem(3703, "https://github.com/dotnet/roslyn-analyzers/issues/3703")]
+        public async Task CSharpAwaitForEachConfigureAwaitWithCancellation()
+        {
+            var code = @"
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+public static class AsyncExtensions
+{
+    public static async Task<int> Sum(this IAsyncEnumerable<int> items, CancellationToken cancellationToken)
+    {
+        int sum = 0;
+        // ConfigureAwait/WithCancellation can be in either order.
+        await foreach (var item in items.ConfigureAwait(false).WithCancellation(cancellationToken))
+            sum += item;
+        return sum;
+    }
+}
+";
+
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Default.AddPackages(
+                    ImmutableArray.Create(new PackageIdentity("Microsoft.Bcl.AsyncInterfaces", "5.0.0"))),
```

Comment:
❔  Can we use `ReferenceAssemblies.NetCore.NetCoreApp31` instead?

Reply:
Microsoft.Bcl.AsyncInterfaces is the smallest additional package to support IAsyncEnumerable, and even works on .net framework.
It also matches the package you used for the 'await using' unit tests.